packages/ember-data/lib/system/snapshot.js

```diff
@@ -392,6 +392,16 @@ Snapshot.prototype = {
   },
 
   /**
+    @method serialize
+    @param {Object} options
+    @return {Object} an object whose values are primitive JSON values only
+   */
+  serialize: function(options) {
+    return this.record.store.serializerFor(this.modelName).
```

Comment:
good point.
Can you then make store.serialize use `snapshot.serialize` instead of repeating code?